traffic_ops/build/traffic_ops.spec

```diff
@@ -131,7 +131,7 @@ Built: %(date) by %{getenv: USER}
 
     # upgrade
     if [ "$1" == "2" ]; then
-	systemctl stop traffic_ops
+        systemctl stop traffic_ops || true # Don't exit with an error if we are in an unprivileged Docker container and can't use systemctl
```

Comment:
I have some concerns about this in production systems. What would happen if a production upgrade of Traffic Ops fails to stop from the `systemctl stop traffic_ops`? 

I'm concerned if an upgrade is scripted, and automation tools log or alarm on a nonzero exit code. Even if it isn't a script, humans might reasonably expect a 0 exit code to mean success.

Besides the exit code, that's going to make this script continue to run, even if `systemctl stop` fails. What happens when the script keeps going, while TO is running? Does it succeed and keep running the old TO? Does it fail elsewhere in an easy to debug way? Is whatever happens deterministic?

It might be totally fine, I'd just like to be sure before we do this where it'll be installed in Prod. Especially since Mojo has a bad habit of refusing to die when told to.

If the goal is primarily to fix #4249 I think we can probably just change the CiaB Dockerfiles to `FROM centos/systemd`; IMO we should probably do that anyway. It's valuable to make the CiaB emulate Prod as closely as possible, for testing.

Reply:
I understand concerns about just silently exiting if the service itself fails to stop. What do you think about skipping that attempt to stop the service if systemd itself does not seem operational? E.g.,

```shell
if [ "$1" == "2" ] && systemctl list-jobs; then
    systemctl stop traffic_ops
fi
```

Reply:
>What do you think about skipping that attempt to stop the service if systemd itself does not seem operational?

That sounds like a good idea to me. Would you object to adding a tiny bit more logic, to log a warning if that happens? E.g.

```sh
if [ "$1" == "2" ]
  if ! systemctl list-jobs; then
    systemctl stop traffic_ops
  else
    echo 'WARNING: systemctl failed - not stopping Traffic Ops!' > /dev/stderr
  fi
fi
```

Reply:
That sounds fine.

Reply:
Addressed in 7a356ace